debezium/src/main/scala/akka/stream/alpakka/debezium/CommittableRecord.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.debezium
+
+import akka.Done
+import akka.actor.ActorRef
+import org.apache.kafka.connect.source.SourceRecord
+import org.apache.kafka.connect.storage.Converter
+
+final class CommittableRecord(val record: SourceRecord,
+                              private[debezium] val committer: ActorRef,
+                              private[debezium] val converter: Converter)
+    extends DebeziumRecord {
+
+  private[debezium] def markAsProcessed(): Done = {
+    committer ! record
+    Done
```

Comment:
The thing is: I was using the ask pattern there, but, actually it makes no difference since `markProcessed` and `markBatchFinished` just marks the message to be committed and not really commit it.
Also, what's preventing back-pressure to work properly is actually how DebeziumEngine works: it expects a blocking `notifying` function. No matter how we commit offsets, it has nothing to do with Debezium back-pressuring whatsoever. So, if commit or not commit anything, Debezium will keep to emit new notifications.
I've done a few changes and I forgot to push, sorry, I'm going to push it and so you can take a look at it. This way I was able to simulate a "fair enough" back-pressure.

Reply:
That's interesting. I see your `notifiying` implementation now.